### PR TITLE
Small fix (escaping groups when using grep)

### PR DESCRIPTION
**Pull request by Unknownstaffmembe**
Had some problems with notifications and it turns out groups in grep weren't escaped, grep uses basic regular expressions by default; `(` and `)` are treated as regular characters and need to be escaped if they're going to be used for a group.

### CODE REVIEW

**Comment by VHSgunzo** [comment]
@Unknownstaffmembe 
Describe your problem in more detail, now I see that your commit will break everything.
image

**Comment by Unknownstaffmembe** [comment]
My apologies, escaped every group in every grep regular expression and assumed it was all fine since notifications worked; have attached an image of the commands being run, it appears that grep uses basic regular expressions by default on my system, perhaps doing grep -E would be better than manually escaping groups and, who doesn't seem to output the display number for X either (most likely due to who and w being separate utilities on my system).
The bottom line shows a workaround but, I can't test whether this is indeed portable.
